src/lib.rs

```diff
@@ -188,25 +202,32 @@ where
     ///     let mut device = AstarteDeviceSdk::new(sdk_options).await.unwrap();
     /// }
     /// ```
-    pub async fn new(opts: AstarteOptions<S>) -> Result<AstarteDeviceSdk<S>, Error> {
+    pub async fn new(
+        opts: AstarteOptions<S>,
+    ) -> Result<(AstarteDeviceSdk<S>, EventReceiver), Error> {
         let mqtt_options = pairing::get_transport_config(&opts).await?;
 
         debug!("{:#?}", mqtt_options);
 
         let (client, eventloop) = AsyncClient::new(mqtt_options, 50);
 
+        let (tx_events, rx_events) = mpsc::channel(50);
```

Comment:
If `50` refers to the same think (specifies the capacity of the bounded channel) maybe should be defined as constant.
```suggestion
        let (client, eventloop) = AsyncClient::new(mqtt_options, MQTT_CHANNEL_SIZE);

        let (tx_events, rx_events) = mpsc::channel(MQTT_CHANNEL_SIZE);
```

Reply:
Done